orc8r/cloud/go/syncstore/reader.go

```diff
@@ -0,0 +1,245 @@
+/*
+ Copyright 2020 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package syncstore
+
+import (
+	"database/sql"
+	"encoding/binary"
+
+	"magma/orc8r/cloud/go/blobstore"
+	configurator_storage "magma/orc8r/cloud/go/services/configurator/storage"
+	"magma/orc8r/cloud/go/sqorc"
+	"magma/orc8r/cloud/go/storage"
+	merrors "magma/orc8r/lib/go/errors"
+	"magma/orc8r/lib/go/protos"
+
+	"github.com/Masterminds/squirrel"
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+)
+
+type syncStoreReader struct {
+	db      *sql.DB
+	builder sqorc.StatementBuilder
+	fact    blobstore.BlobStorageFactory
+}
+
+func NewSyncStoreReader(db *sql.DB, builder sqorc.StatementBuilder, fact blobstore.BlobStorageFactory) SyncStoreReader {
+	return &syncStoreReader{db: db, builder: builder, fact: fact}
+}
+
+func (l *syncStoreReader) Initialize() error {
```

Comment:
My concern with replacing `repeated LeafDigest` in other messages with `LeafDigests` was that it will result in an extra layer of data abstraction, and we'd have to unpack an extra data object to get to the list of leaf digests. Given that, not sure if I should choose consistency (using `LeafDigests` everywhere) over simplicity (directly use `repeated LeafDigest` in `DigestTree`)? @hcgatewood